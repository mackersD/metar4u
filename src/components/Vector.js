import React from 'react'

const Vector = (props) => {

  var direction = props.direction ? props.direction : 0
  var distance = props.distance ? props.distance : 0

  var arrowStyle = {
    "msTransform": "rotate(" + direction + "deg)",
    "WebkitTransform": "rotate(" + direction + "deg)",
    "transform": "rotate(" + direction + "deg)",
    "display": "inline-block"
  }

  return (
    <div className="vector">
      <div className="vectorArrow" style={arrowStyle}>{"^"}</div>
      <div className="vectorDistance">{Math.round(distance)}</div>
    </div>
  )
}

export default Vector
