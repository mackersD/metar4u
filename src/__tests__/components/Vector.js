import React from 'react'
import renderer from 'react-test-renderer'
import Vector from '../../components/Vector'

it('Vector - renders', () => {
  const tree = renderer.create(
    <Vector />
  )
  expect(tree).toMatchSnapshot()
})

it('Vector - positive rotation', () => {
  var tree = renderer.create(
    <Vector
      direction={45}
    />
  )
  expect(tree).toMatchSnapshot()

  tree = renderer.create(
    <Vector
      direction={540}
    />
  )
  expect(tree).toMatchSnapshot()
})

it('Vector - negative rotation', () => {
  var tree = renderer.create(
    <Vector
      direction={-45}
    />
  )
  expect(tree).toMatchSnapshot()

  tree = renderer.create(
    <Vector
      direction={-540}
    />
  )
  expect(tree).toMatchSnapshot()
})

it('Vector - distance', () => {
  var tree = renderer.create(
    <Vector
      distance={20}
    />
  )
  expect(tree).toMatchSnapshot()

  tree = renderer.create(
    <Vector
      distance={-20}
    />
  )
  expect(tree).toMatchSnapshot()
})

it('Vector - direction + distance', () => {
  const tree = renderer.create(
    <Vector
      direction={90}
      distance={20}
    />
  )
  expect(tree).toMatchSnapshot()
})
