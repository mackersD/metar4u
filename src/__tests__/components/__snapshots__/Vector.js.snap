// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vector - direction + distance 1`] = `
<div
  className="vector"
>
  <div
    className="vectorArrow"
    style={
      Object {
        "WebkitTransform": "rotate(90deg)",
        "display": "inline-block",
        "msTransform": "rotate(90deg)",
        "transform": "rotate(90deg)",
      }
    }
  >
    ^
  </div>
  <div
    className="vectorDistance"
  >
    20
  </div>
</div>
`;

exports[`Vector - distance 1`] = `
<div
  className="vector"
>
  <div
    className="vectorArrow"
    style={
      Object {
        "WebkitTransform": "rotate(0deg)",
        "display": "inline-block",
        "msTransform": "rotate(0deg)",
        "transform": "rotate(0deg)",
      }
    }
  >
    ^
  </div>
  <div
    className="vectorDistance"
  >
    20
  </div>
</div>
`;

exports[`Vector - distance 2`] = `
<div
  className="vector"
>
  <div
    className="vectorArrow"
    style={
      Object {
        "WebkitTransform": "rotate(0deg)",
        "display": "inline-block",
        "msTransform": "rotate(0deg)",
        "transform": "rotate(0deg)",
      }
    }
  >
    ^
  </div>
  <div
    className="vectorDistance"
  >
    -20
  </div>
</div>
`;

exports[`Vector - negative rotation 1`] = `
<div
  className="vector"
>
  <div
    className="vectorArrow"
    style={
      Object {
        "WebkitTransform": "rotate(-45deg)",
        "display": "inline-block",
        "msTransform": "rotate(-45deg)",
        "transform": "rotate(-45deg)",
      }
    }
  >
    ^
  </div>
  <div
    className="vectorDistance"
  >
    0
  </div>
</div>
`;

exports[`Vector - negative rotation 2`] = `
<div
  className="vector"
>
  <div
    className="vectorArrow"
    style={
      Object {
        "WebkitTransform": "rotate(-540deg)",
        "display": "inline-block",
        "msTransform": "rotate(-540deg)",
        "transform": "rotate(-540deg)",
      }
    }
  >
    ^
  </div>
  <div
    className="vectorDistance"
  >
    0
  </div>
</div>
`;

exports[`Vector - positive rotation 1`] = `
<div
  className="vector"
>
  <div
    className="vectorArrow"
    style={
      Object {
        "WebkitTransform": "rotate(45deg)",
        "display": "inline-block",
        "msTransform": "rotate(45deg)",
        "transform": "rotate(45deg)",
      }
    }
  >
    ^
  </div>
  <div
    className="vectorDistance"
  >
    0
  </div>
</div>
`;

exports[`Vector - positive rotation 2`] = `
<div
  className="vector"
>
  <div
    className="vectorArrow"
    style={
      Object {
        "WebkitTransform": "rotate(540deg)",
        "display": "inline-block",
        "msTransform": "rotate(540deg)",
        "transform": "rotate(540deg)",
      }
    }
  >
    ^
  </div>
  <div
    className="vectorDistance"
  >
    0
  </div>
</div>
`;

exports[`Vector - renders 1`] = `
<div
  className="vector"
>
  <div
    className="vectorArrow"
    style={
      Object {
        "WebkitTransform": "rotate(0deg)",
        "display": "inline-block",
        "msTransform": "rotate(0deg)",
        "transform": "rotate(0deg)",
      }
    }
  >
    ^
  </div>
  <div
    className="vectorDistance"
  >
    0
  </div>
</div>
`;
